ee/clickhouse/models/action.py

```diff
@@ -61,27 +62,24 @@ def format_action_filter(
 def filter_event(
     step: ActionStep, prepend: str = "event", index: int = 0, table_name: str = ""
 ) -> Tuple[List[str], Dict]:
+    from ee.clickhouse.models.property import get_property_string_expr
+
     params = {"{}_{}".format(prepend, index): step.event}
     conditions = []
 
     if table_name != "":
         table_name += "."
 
     if step.url:
+        value_expr, _ = get_property_string_expr("events", "$current_url", "'$current_url'", f"{table_name}properties")
         if step.url_matching == ActionStep.EXACT:
-            conditions.append(
-                f"JSONExtractString({table_name}properties, '$current_url') = %({prepend}_prop_val_{index})s"
-            )
+            conditions.append(f"{value_expr} = %({prepend}_prop_val_{index})s")
```

Comment:
minor nit: Can a variable replace `{prepend}_prop_val_{index}` in all 4 places here? Easier to read + harder to make mistakes.